lib/core/services/analytics_service.dart

```diff
@@ -9,15 +10,21 @@ class AnalyticsService {
 
   final FirebaseAnalytics _analytics = FirebaseAnalytics();
 
+  final FirebaseCrashlytics _crashlytics = FirebaseCrashlytics.instance;
+
   FirebaseAnalyticsObserver getAnalyticsObserver() =>
       FirebaseAnalyticsObserver(analytics: _analytics);
 
   /// Log a error. [prefix] should be the service where the error was triggered.
-  Future logError(String prefix, String message) async {
+  Future logError(String prefix, String message, [Exception error]) async {
     final mesTruncated =
         message.length > 100 ? message.substring(0, 99) : message;
     await _analytics.logEvent(
         name: "${prefix}Error", parameters: {'message': mesTruncated});
+
+    if (error != null) {
+      await _crashlytics.recordError(error, null, reason: message);
```

Comment:
Could it be useful to pass the stacktrace as the second argument to recordError? It may be redundant but it could in some case, help pinpoint the error in code afterward? (Though maybe it could floud crashlytics with a lot of data which may be not useful, as I don't know crashlytics behavior this is just an idea)😄

Reply:
You have completely right! :sweat_smile: